black_hole/discord.py

```diff
@@ -138,9 +152,16 @@ async def bridge(self, room, msg, member, source):
 
         log.debug("adding message to queue")
 
+        reply_message_id = (
+            None if msg.xep0308_replace is None else msg.xep0308_replace.id_
+        )
+
         # add this message to the queue
         self._queue.append(
             {
+                "author_jid": str(member.direct_jid),
+                "upstream_message_id": msg.id_,
```

Comment:
i'm not too sure about xmpp semantics but are messages guaranteed to have ids?

Reply:
also, it might be better to replace `upstream` with `xmpp`?

Reply:
[It doesn't look like it](https://docs.zombofant.net/aioxmpp/devel/api/public/stanza.html?highlight=id_#aioxmpp.Message.id_), interesting.

Reply:
i think this pr should be updated to take account of that then :thonk: